h2/src/main/org/h2/mvstore/MVStore.java

```diff
@@ -292,12 +292,21 @@ to a map (possibly the metadata map) -
         Object o = config.get("compress");
         this.compressionLevel = o == null ? 0 : (Integer) o;
         String fileName = (String) config.get("fileName");
+        fileStore = (FileStore) config.get("fileStore");
+        fileStoreIsProvided = fileStore != null;
+        if(fileStore == null && fileName != null) {
+            fileStore = new FileStore();
+        }
         o = config.get("pageSplitSize");
-        if (o == null) {
-            pageSplitSize = fileName == null ? 4 * 1024 : 16 * 1024;
+        int pgSplitSize;
+        if (o != null) {
+            pgSplitSize = (Integer) o;
+        } else if(fileStore != null) {
+            pgSplitSize = 16 * 1024;
         } else {
-            pageSplitSize = (Integer) o;
+            pgSplitSize = 48; // number of keys per page in that case
```

Comment:
Yes, it was intentional, since in that case accounting is in terms of keys per page, and 48 looks like sweet spot between big pages (short tree, effective binary searches, but high cost of cloning) and small ones (fast to clone pages, but tall tree with high search overhead).

@thomasmueller, what do you think about more traditional accounting - in terms of branching order (keys-per-page count) instead of expensive page memory size calculation? If we are not trying to make pages uniform in size anymore in that new SSD world, then why bother? That said, we still need a rough page size estimate, as it's required by the cache management, but this can be derived from a table stats, i.e. Any thoughts?
Another question about MVStore: any reason why re-balancing on deletion (keys redistribution / node merge) is not implemented?